windows/src/engine/keyman32/selectkeyboard.cpp

```diff
@@ -179,23 +179,42 @@ void SelectKeyboardHKL(PKEYMAN64THREADDATA _td, DWORD hkl, BOOL foreground) {
     }
 }
 
+void PrepareLanguageSwitchString(UINT langid, HKL hkl, char *str) {
+  wsprintf(str, "%d|%d|%d", GetCurrentThreadId(), langid, hkl);   // I4285
+}
+
+void PrepareLanguageSwitchString(UINT langid, GUID clsid, GUID guidProfile, char *str) {
+  WCHAR clsidstr[40], profilestr[40];
```

Comment:
Is the 40 character limit tied to anything?

Reply:
38 characters in a GUID string returned from StringFromGUID2. https://stackoverflow.com/questions/968175/what-is-the-string-length-of-a-guid (brace format)